network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/snooze/VpnDisableOnCall.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.snooze
+
+import android.content.Context
+import android.telephony.PhoneStateListener
+import android.telephony.TelephonyManager
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.Vpn
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlin.properties.Delegates
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import logcat.LogPriority
+import logcat.asLog
+import logcat.logcat
+
+interface VpnDisableOnCall {
+    fun enable()
+    fun disable()
+    fun isCallInProgress(): Boolean
+}
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class RealVpnDisableOnCall @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val context: Context,
+    private val vpn: Vpn,
+) : VpnDisableOnCall {
+    private var _callInProgress: Boolean = false
+    private val telephonyManager by lazy {
+        context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager?
+    }
+    private val _listener: PhoneStateListener =
+        object : PhoneStateListener() {
+            @Deprecated("Deprecated in Java")
+            override fun onCallStateChanged(
+                state: Int,
+                phoneNumber: String?,
+            ) {
+                appCoroutineScope.launch(dispatcherProvider.io()) {
+                    logcat { "Call state: $state" }
+                    if (state == TelephonyManager.CALL_STATE_IDLE) {
+                        _callInProgress = false
+                        vpn.start()
```

Comment:
The downside of using VPN here is that IF user has both AppTP and NetP enabled AND this setting is enabled, once the VPN subscription expires, user will not be able to turn off this setting.

Reply:
Should we add a check `networkProtectionState.isEnabled()` in the `else` branch?